public/app/plugins/datasource/elasticsearch/datasource.js

```diff
@@ -124,11 +128,12 @@ function (angular, _, moment, kbn, ElasticQueryBuilder, IndexPattern, ElasticRes
 
         for (var i = 0; i < hits.length; i++) {
           var source = hits[i]._source;
-          var fields = hits[i].fields;
           var time = source[timeField];
-
-          if (_.isString(fields[timeField]) || _.isNumber(fields[timeField])) {
-            time = fields[timeField];
+          if (typeof hits[i].fields !== 'undefined') {
```

Comment:
Should this be `if (this.esVersion < 5) {` for consistency since the above `// fields field not supported on ES 5.x` change means we're no longer requesting fields ES >= 5?